src/test/java/com/puppycrawl/tools/checkstyle/grammar/AstRegressionTest.java

```diff
@@ -246,6 +262,30 @@ public void testImpossibleValid() throws Exception {
                 '4', '4', '.', '4', 'P', '4', 'D', ';');
     }
 
+    @Test
+    public void testImpossibleExceptionsJava14TextBlockLexer() throws Exception {
+        AssertGeneratedJava14TextBlockLexer.verifyFail("mSTD_ESC", '\\', '*', (char) 0xFFFF);
+        AssertGeneratedJava14TextBlockLexer.verifyFail("mONE_DOUBLE_QUOTE", '"', '"');
+        AssertGeneratedJava14TextBlockLexer.verifyFail("mNEWLINE", '*');
+        AssertGeneratedJava14TextBlockLexer.verifyFail("mTEXT_BLOCK_CONTENT", (char) 0xFFFF);
+    }
+
+    @Test
+    public void testImpossibleValidJava14TextBlockLexer() throws Exception {
```

Comment:
I am getting complaints from both IDEA and TeamCity about this: ` JUnit test method testImpossibleExceptionsJava14TextBlockLexer() contains no assertions` which I understand, but why don't the existing tests (see [here](https://github.com/checkstyle/checkstyle/blob/51836b52b04c5681a34254b09b78364a04be87dc/src/test/java/com/puppycrawl/tools/checkstyle/grammar/AstRegressionTest.java#L226)) cause this warning?

Reply:
This means either there is a suppression in the config, or somewhere it is specified that AssertGeneratedJavaLexer.verifyFail is an assertion method.

A quick search finds these rules:
https://github.com/checkstyle/checkstyle/blob/master/config/pmd-test.xml#L182
https://github.com/checkstyle/checkstyle/blob/master/config/intellij-idea-inspections.xml#L4618
(the list may be incomplete)